tools/app/app/cards/layout.tsx

```diff
@@ -24,18 +25,28 @@ import {
   THEME_ID as MATERIAL_THEME_ID,
 } from '@mui/material/styles'
 import { CssVarsProvider as JoyCssVarsProvider } from '@mui/joy/styles'
-import NewCardDialog from '../components/NewCardDialog'
-import { useTemplates, useCard } from '../lib/api'
-import ErrorBar from '../components/ErrorBar'
-import { useCardKey, useError } from '../lib/utils'
+import { NewCardModal } from '../components/modals'
+import StoreProvider from '../providers/StoreProvider'
+import { useAppDispatch, useAppSelector, useCardKey } from '../lib/hooks'
+import { CloseRounded } from '@mui/icons-material'
+import {
+  closeNotification,
+  removeNotification,
+} from '../lib/slices/notifications'
 import { useRouter } from 'next/navigation'
-import { useTranslation } from 'react-i18next'
 
 function AppLayout({ children }: Readonly<{ children: React.ReactNode }>) {
   // Last URL parameter after /cards base is the card key
   const urlCardKey = useCardKey()
```

Comment:
Should be able to get card key from params.key if it is available in the route.

https://nextjs.org/docs/app/api-reference/file-conventions/layout#params-optional